pkg/queue/postgres/setup.go

```diff
@@ -0,0 +1,102 @@
+package postgres
+
+import (
+	"bytes"
+	"context"
+	"text/template"
+
+	"github.com/contiamo/go-base/pkg/db"
+)
+
+type ForeignReference struct {
+	ColumnName       string
+	ColumnType       string
+	ReferencedTable  string
+	ReferencedColumn string
+}
+
+func Setup(ctx context.Context, db db.SQLDB, references []ForeignReference) error {
+	buf := &bytes.Buffer{}
+	err := dbSetupTemplate.Execute(buf, references)
+	if err != nil {
+		return err
+	}
+	_, err = db.ExecContext(ctx, buf.String())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+var dbSetupTemplate = template.Must(template.New("queue-db-setup").Parse(`
+CREATE EXTENSION IF NOT EXISTS citext;
+
+CREATE TABLE IF NOT EXISTS schedules (
+      schedule_id uuid PRIMARY KEY,
+    task_queue citext NOT NULL,
+    task_type citext NOT NULL,
+    task_spec jsonb NOT NULL,
+    cron_schedule citext NOT NULL DEFAULT '',
+    next_execution_time timestamptz,
+    created_at timestamptz NOT NULL DEFAULT NOW(),
+    updated_at timestamptz NOT NULL DEFAULT NOW()--,
```

Comment:
This piece looks a bit confusing.
```suggestion
    updated_at timestamptz NOT NULL DEFAULT NOW()
```

Reply:
done